integration/update_field_compat_test.go

```diff
@@ -666,9 +666,24 @@ func TestUpdateFieldCompatSet(t *testing.T) {
 			update:        bson.D{{"$set", bson.D{{"foo", int32(12)}, {"v", nil}}}},
 			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/1676",
 		},
-		"SetSameValueInt": {
-			update: bson.D{{"$set", bson.D{{"v", int32(42)}}}},
-			skip:   "https://github.com/FerretDB/FerretDB/issues/1662",
+		"SameValueInt32Type": {
+			update:        bson.D{{"$set", bson.D{{"v", int32(42)}}}},
+			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/1668",
+		},
+		"SameValueInt64Type": {
+			update:        bson.D{{"$set", bson.D{{"v", int64(42)}}}},
+			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/1668",
+		},
+		"SameValueDoubleType": {
+			update:        bson.D{{"$set", bson.D{{"v", 42.0}}}},
+			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/1668",
```

Comment:
Same, as far as I understand it should work for Tigris

Reply:
For the same as above, this can only be tested on double data, I updated the test.